submissions/akulahere/frogger-game/app.js

```diff
@@ -0,0 +1,93 @@
+const playerStartX = 200;
+const playerStartY = 400;
+const stepX = 100;
+const stepY = 85;
+const finishLineY = 0;
+const rightBorder = 500;
+const enemyStartX = 0;
+const enemyMinSpeed = 100;
+const enemyMaxSpeed = 400;
+const enemyStartYPositions = [60, 145, 230]
+
+function getRandomInteger(min, max) {
+  return Math.floor(Math.random() * (max - min + 1)) + min;
+}
+
+function makeRound100(val) {
+  return Math.round(val / 100) * 100;
+}
+
+var Enemy = function (y) {
+  this.x = enemyStartX;
+  this.y = y;
+  this.speed = getRandomInteger(enemyMinSpeed, enemyMaxSpeed);
+  this.sprite = 'images/enemy-bug.png';
+};
+
+Enemy.prototype.update = function (dt) {
+  this.x += this.speed * dt;
+  if (this.x > rightBorder) {
+    this.x = enemyStartX;
+  }
+};
+
+Enemy.prototype.render = function () {
+  ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+
+let Player = function (x,y) {
+  this.x = x;
+  this.y = y;
+  this.sprite = 'images/char-boy.png';
+}
+
+Player.prototype.update = function () {
+  allEnemies.forEach(enemy => {
```

Comment:
Classes should not refer to any global variables, like global variable `player`, which is an instance of `Player` class, or `allEnemies`. Handle those just like any other values from the world outside `Player` or `Enemy`.

Referring to global constants and globals provided by the gaming platform like `Resources` is OK.

Reply:
Also this method does two different things:
1. Updates a player's position (as method name promises)
2. Checks for collisions (method name doesn't promise this)

This breaks Single Responsibility Principle. Please resolve.

Reply:
Done.
isCollision function added and update method refactored.